### PR TITLE
Duplicate filter tolerance

### LINKED ISSUE #150: Filter Subprovider Adds Redundant Filters
Multiple calls to `newFilter` with the same parameters will create a new polling process, even if the new filter has identical params.

We should dedupe redundant filters, to avoid creating filter leaks like this.

I think this is a pretty solid first mitigation strategy for https://github.com/MetaMask/eth-block-tracker/issues/21 because even if Web3 doesn't fix their Filter leak, we'll only leak as many filters as get registered by sites the user visits, which have a chance at being much more sustainable.

### PR DESCRIPTION
**Pull request by flyswatter**
Fixes #150 

When adding a filter, now checks for existent identical filters, and returns those filter IDs instead of new ones if they exist.

No longer creates redundant filters as `Event.get()` is called.

Also fixed some tests that were intolerant of our newly enforced zero-prefix stripping on numeric (non-blob) values, as well as a bug in the filter test generator that prevented the filter subprovider from responding to new blocks before its filters were tested.

@kumavis

### CODE REVIEW

File: test/filters.js [hunk 1]
```diff
@@ -211,7 +211,10 @@ function filterTest(label, filterPayload, afterInstall, filterChangesOne, filter
     engine.addProvider(filterProvider)
     engine.addProvider(blockProvider)
     engine.once('block', startTest)
-    engine.start()
+
+    setTimeout(() => {
+      engine.start()
+    }, 1)
```
Comment (flyswatter) [inline test/filters.js]: Sometimes the `startTest` would be fired, and would check the filter subprovider for recognizing new filters before the filter subprovider had received the `block` event, which it uses to identify filter changes.

This timeout allows the filter subprovider an opportunity to cache updates to return on subsequent requests.

File: test/util/block.js [hunk 2]
```diff
@@ -72,7 +71,7 @@ TestBlockProvider.prototype.addTx = function(txParams){
 function createBlock(blockParams, prevBlock, txs) {
   blockParams = blockParams || {}
   txs = txs || []
-  var defaultNumber = prevBlock ? incrementHex(prevBlock.number) : '0x01'
+  var defaultNumber = prevBlock ? incrementHex(prevBlock.number) : '0x1'
```
Comment (flyswatter) [inline test/util/block.js]: Making our block numbering consistently compliant with geth's new scheme makes our tests deterministic again.

**Comment by kumavis** [comment]
@flyswatter if we:
- install filter A
- install filter B
- uninstall filter A
is filter B uninstalled?

**Comment by flyswatter** [comment]
Damn, probably.  I guess we should keep single filters, but have multiple IDs mapping to them?
